### PR TITLE
Create set_build_number_repository.rb

### PR DESCRIPTION
**Pull request by pbrooks**
This new action allows the build number to be set from the repository (currently supports svn, git-svn and git)

I'm not a ruby programmer, so please feel free to pull this apart and reimplement (I won't be offended :)), we just needed this functionality quickly.

### CODE REVIEW

File: lib/fastlane/set_build_number_repository.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  module Actions
+    module SharedValues
+    end
+
+    class SetBuildNumberRepositoryAction < Action
+      def self.is_supported?(platform)
+        platform == :ios
+      end
+
+      def self.is_svn?()
```
Comment (KrauseFx) [inline lib/fastlane/set_build_number_repository.rb]: You can drop the `()`

File: lib/fastlane/set_build_number_repository.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      def self.is_git?
+        begin
+          Actions.sh 'git rev-parse HEAD'
+          return true
+        rescue
+          return false
+        end
+      end
+
+      def self.is_git_svn?()
```
Comment (KrauseFx) [inline lib/fastlane/set_build_number_repository.rb]: Same here

**Comment by KrauseFx** [comment]
@pbrooks thanks for submitting the pull request. Could you please move the new action into the `actions` folder?

If you need this now, just copy your action into your local `fastlane/actions` folder and you can already use it :+1: 

The next `fastlane` will be in a few days probably, so better use it as a local action for now.

**Comment by pbrooks** [comment]
Should be sorted now :)

**Comment by KrauseFx** [comment]
Thanks, looks great :+1: Only thing missing (sorry I forgot to mention this before), could you add your new action to [Actions.md](https://github.com/KrauseFx/fastlane/blob/master/docs/Actions.md#using-git) in the `Using git` section? 

Thanks again for your work :+1:

**Comment by pbrooks** [comment]
Done :)

**Comment by KrauseFx** [comment]
Thank you @pbrooks, this will be in the next release. As mentioned above if you need this action right now, please use it locally for now, as I won't release a new version today.

**Pull request merged**